application/apps/indexer/addons/file-tools/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+use anyhow::Result;
+use std::{
+    fs::{metadata, File},
+    io::Read,
+    path::Path,
+    str::from_utf8,
+};
+
+const BYTES_TO_READ: u64 = 10240;
+
+pub fn is_binary(file_path: String) -> Result<bool> {
+    let chunks = fetch_starting_chunk(Path::new(&file_path));
+    let buffer = match chunks {
+        Ok(buffer) => buffer,
+        Err(err) => return Err(err),
+    };
+
+    let result = from_utf8(&buffer);
+    match result {
+        Ok(_file_content) => Ok(false),
+        Err(_err) => Ok(true),
+    }
+}
+
+fn fetch_starting_chunk(file_path: &Path) -> Result<Vec<u8>> {
+    let file = File::open(file_path)?;
+    let file_length: u64 = metadata(file_path)?.len();
+    let file_length: u64 = if file_length == 0 {
+        file_length
+    } else {
+        file_length - 1
+    };
+    let file_length = if BYTES_TO_READ < file_length {
+        BYTES_TO_READ
+    } else {
+        file_length
+    };
+
+    let mut file = file.take(file_length);
+    let mut buffer = Vec::new();
+    file.read_to_end(&mut buffer)?;
+    Ok(buffer)
+}
+
```

Comment:
This is right. And this is work. I do not mind if this block will no be changed, but it still can be visually optimized:
```
fn fetch_starting_chunk(file_path: &Path) -> Result<Vec<u8>> {
    let bytes_to_read: u64 = (metadata(file_path)?.len().max(1) - 1).min(BYTES_TO_READ);
    let mut buffer = Vec::new();
    File::open(file_path)?
        .take(bytes_to_read)
        .read_to_end(&mut buffer)?;
    Ok(buffer)
}
```